airflow/providers/amazon/aws/operators/ecs.py

```
@@ -452,6 +460,11 @@ def _check_success_task(self) -> None:
         if len(response.get('failures', [])) > 0:
             raise AirflowException(response)
 
+        if self.task_log_fetcher and not self.task_log_fetcher.is_log_stream_created():
+            raise AirflowException(
+                'Cloudwatch log stream for the task is not found. The task probably failed to start.'
+            )
+
         for task in response['tasks']:
             # This is a `stoppedReason` that indicates a task has not
             # successfully finished, but there is no other indication of failure
```

aha, we do check `stoppedReason` here. but the "Host EC2 (instance id) stopped/terminated" is a much rarer reason than "Timeout waiting for network interface provisioning to complete." according to AWS support when i asked about "common" reasons for silently failed tasks

---

Yes, but in our case we have a clear `stopCode` that has to be handled. So it's easier. In that place we can have some edge case handlers, i think that's totally fine.

---

:+1: